### PR TITLE
Move build to GitHub actions

### PR DESCRIPTION
**Pull request by malmckay**
Drops TravisCI (which is no longer free for opensource)
Drops MRI Ruby < 2.5, JRuby < latest, and other rubies

Adds Github Actions builder, set to GCC 7 (we'll add GCC latest with a fix for that version)

### CODE REVIEW

File: ext/c_zookeeper.rb [hunk 1]
```diff
@@ -328,7 +328,7 @@ def submit_pending_calls
     end
 
     def wake_event_loop!
-      @pipe_write && @pipe_write.write('1')
+      @pipe_write && !@pipe_write.closed? && @pipe_write.write('1')
```
Comment (nerdrew) [inline ext/c_zookeeper.rb:331]: Is this change intentional?

**Review by nerdrew — APPROVED** [review]

File: ext/c_zookeeper.rb (hunk 1 above)
Comment (malmckay) [reply ext/c_zookeeper.rb:331]: This is a fix for a flaky test, eg https://github.com/malmckay/zookeeper/runs/3675238341?check_suite_focus=true
